azure-iot-device/tests/common/test_mqtt_transport.py

```diff
@@ -170,13 +133,49 @@ def test_configures_tls_context_with_client_provided_certificate_chain(
             fake_client_cert.pass_phrase,
         )
 
+    @pytest.mark.it("Sets Paho MQTT Client callbacks")
+    def test_sets_paho_callbacks(self, mocker):
+        mock_mqtt_client = mocker.patch.object(mqtt, "Client").return_value
+
+        MQTTTransport(client_id=fake_device_id, hostname=fake_hostname, username=fake_username)
+
+        assert callable(mock_mqtt_client.on_connect)
+        assert callable(mock_mqtt_client.on_disconnect)
+        assert callable(mock_mqtt_client.on_subscribe)
+        assert callable(mock_mqtt_client.on_unsubscribe)
+        assert callable(mock_mqtt_client.on_publish)
+        assert callable(mock_mqtt_client.on_message)
+
+    @pytest.mark.it("Initializes event handler callbacks to 'None'")
+    def test_handler_callbacks_set_to_none(self, mocker):
+        mocker.patch.object(mqtt, "Client")
+
+        transport = MQTTTransport(
+            client_id=fake_device_id, hostname=fake_hostname, username=fake_username
+        )
+
+        assert transport.on_mqtt_connected is None
+        assert transport.on_mqtt_disconnected is None
+        assert transport.on_mqtt_message_received is None
+
+    @pytest.mark.it("Initializes internal operation tracking structures")
+    def test_operation_infrastructure_set_up(self, mocker):
+        transport = MQTTTransport(
+            client_id=fake_device_id, hostname=fake_hostname, username=fake_username
+        )
+        assert transport._op_manager._pending_operation_callbacks == {}
+        assert transport._op_manager._unknown_operation_completions == {}
+
+
+@pytest.mark.describe("MQTTTransport - .connect()")
+class TestConnect(object):
     @pytest.mark.it("Sets username and password")
```

Comment:
>Sets [](start = 21, length = 4)

I noticed these tests aren't sufficiently testing the optionality of the password. These could use some shoring up (same for reconnect). I'll be doing this before I merge this.